internal/logger/io_sink.go

```diff
@@ -82,14 +212,22 @@ func (osSink *IOSink) Info(_ int, msg string, keysAndValues ...interface{}) {
 		kvMap[keysAndValues[i].(string)] = keysAndValues[i+1]
 	}
 
-	switch msg {
-	case CommandStarted:
-		logCommandMessageStarted(osSink.log, kvMap)
-	case CommandSucceeded:
-		logCommandMessageSucceeded(osSink.log, kvMap)
-	case CommandFailed:
-		logCommandMessageFailed(osSink.log, kvMap)
-	}
+	map[string]func(*log.Logger, map[string]interface{}){
+		CommandStarted:            logCommandStartedMessage,
+		CommandSucceeded:          logCommandSucceededMessage,
+		CommandFailed:             logCommandFailedMessage,
+		ConnectionPoolCreated:     logPoolCreatedMessage,
+		ConnectionPoolReady:       logPoolReadyMessage,
+		ConnectionPoolCleared:     logPoolClearedMessage,
+		ConnectionPoolClosed:      logPoolClosedMessage,
+		ConnectionCreated:         logConnectionCreatedMessage,
+		ConnectionReady:           logConnectionReadyMessage,
+		ConnectionClosed:          logConnectionClosedMessage,
+		ConnectionCheckoutStarted: logConnectionCheckoutStartedMessage,
+		ConnectionCheckoutFailed:  logConnectionCheckoutFailed,
+		ConnectionCheckedOut:      logConnectionCheckedOut,
+		ConnectionCheckedIn:       logConnectionCheckedIn,
+	}[msg](osSink.log, kvMap)
```

Comment:
This operation may panic if there is no entry for `msg` or if the entry returns a `nil` function reference. Update it to gracefully handle any missing log message formatting functions, either by skipping the message or using a default message formatter.

E.g. using the updated declaration of the recommend command -> function mapping change above:
```go
func (osSink *IOSink) Info(...) {
	f := commandFunc[msg]
	if f != nil {
		f(osSink.log, kvMap)
		return
	}

	// Appropriate logger not found, use the default message formatter.
	logDefault(osSink.log, kvMap)
}
```